cunumeric/array.py

```diff
@@ -955,12 +955,84 @@ def byteswap(self, inplace=False):
             numpy_array = self.__array__(stacklevel=3).byteswap(inplace=False)
             return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
 
-    @unimplemented
-    def choose(self, choices, out, mode="raise"):
-        numpy_array = self.__array__(stacklevel=3).choose(
-            choices=choices, out=out, mode=mode
+    def choose(self, choices, out=None, mode="raise", stacklevel=2):
+        a = self
+        if out is not None:
+            out = out.convert_to_cunumeric_ndarray(out)
+
+        if isinstance(choices, list):
+            choices = tuple(choices)
+        is_tuple = isinstance(choices, tuple)
+        if is_tuple:
+            n = len(choices)
+            ch_dtype = choices[0].dtype
+            choices = tuple(
+                self.convert_to_cunumeric_ndarray(choices[i]).astype(ch_dtype)
+                for i in range(n)
+            )
+
+        else:
+            choices = self.convert_to_cunumeric_ndarray(choices)
+            #            if self.ndim > choices.ndim - 1:
+            #                for i in range(self.ndim + 1 - choices.ndim):
+            #                    choices = choices[:, None, ...]
+            n = choices.shape[0]
+            ch_dtype = choices.dtype
+            choices = tuple(
+                self.convert_to_cunumeric_ndarray(choices[i, ...])
```

Comment:
The result of an array operation on a cunumeric ndarray is already a cunumeric ndarray, there is no need to convert it.